scripts/commerce.js

```diff
@@ -381,3 +381,31 @@ export async function loadErrorPage(code = 404) {
       document.head.appendChild(newScript);
     });
 }
+
+export function mapProductAcdl(product) {
+  const regularPrice = product?.priceRange?.minimum?.regular?.amount.value
+    || product?.price?.regular?.amount.value || 0;
+  const specialPrice = product?.priceRange?.minimum?.final?.amount.value
+    || product?.price?.final?.amount.value;
+  const currencyCode = product?.priceRange?.minimum?.final?.amount.currency
+    || product?.price?.final?.amount.currency || 'USD';
+  const minimalPrice = product?.priceRange ? regularPrice : undefined;
+  const maximalPrice = product?.priceRange
+    ? product?.priceRange?.maximum?.regular?.amount.value : undefined;
+
+  return {
+    productId: parseInt(product.externalId, 10) || 0,
+    name: product?.name,
+    sku: product?.variantSku || product?.sku,
+    topLevelSku: product?.sku,
+    pricing: {
+      regularPrice,
+      minimalPrice,
+      maximalPrice,
+      specialPrice,
+      currencyCode,
+    },
+    canonicalUrl: new URL(`/products/${product.urlKey}/${product.sku}`, window.location.origin).toString(),
+    mainImageUrl: product?.images?.[0]?.url,
```

Comment:
I believe you will need filter for `roles.includes('main')` to be 100% confident you use the proper image. Or maybe there's an implicit association that the first image is always main.

![image](https://github.com/user-attachments/assets/65451463-585b-413c-b0b4-44f041e21fee)

Reply:
The `main` role might be something specific to a customer as it is not available in the environment we use for the boilerplate.